### PR TITLE
[Admin] Define SatelliteTests

### PR DESCRIPTION
**Pull request by TessaPeruzzi**
Working on #50 - Define SatelliteTests

### CODE REVIEW

**Review by brendanfraser597 — DISMISSED** [review]

File: docs/uml-class-diagrams/admin/test/SatelliteTests/SatelliteTests.puml [hunk 1]
```diff
@@ -2,7 +2,14 @@
...
     class SatelliteTests {
-        
+       + void satNameInIsSatNameOut()
+       + void strengthInIsStrengthOut()
+       + void locInIsLocOut()
+       + void setStrengthIsStrength()
+       + void newLatPointUsedIncrementLat()
+       + void newLngPointUsedIncrementLng()
+       + void newTimeIsIncrementedByHour()
+       + void strengthIsBtwnOneAndTen()
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/admin/test/SatelliteTests/SatelliteTests.puml]: I would add a few more tests for "satNameIsNotNull", "initLocationIsNotNull" and add a case for strengthInIsStrengthOut for the constructor and the setter, and strengthIsBtwnOneAndTen for the constructor and the setter.

File: docs/uml-class-diagrams/admin/test/SatelliteTests/SatelliteTests.puml [hunk 2]
```diff
@@ -2,7 +2,14 @@
 
 package edu.oakland.test.admin {
     class SatelliteTests {
-        
+       + void satNameInIsSatNameOut()
+       + void strengthInIsStrengthOut()
+       + void locInIsLocOut()
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/admin/test/SatelliteTests/SatelliteTests.puml]: I don't think we need this one, as the location that we store in is the initial location, and the location out is the one being generated.

File: docs/uml-class-diagrams/admin/test/SatelliteTests/SatelliteTests.puml [hunk 3]
```diff
@@ -2,7 +2,14 @@
...
+       + void setStrengthIsStrength()
+       + void newLatPointUsedIncrementLat()
+       + void newLngPointUsedIncrementLng()
+       + void newTimeIsIncrementedByHour()
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/admin/test/SatelliteTests/SatelliteTests.puml]: These internal methods will not be tested directly, but will be covered in a test case like getLocation0(), getLocation1(), getLocationN() where you physically call the Satellite.getLocation() method n times, then check to make sure that the last one is incremented as much as it needs to be for that particular point. I've got an example on how to code this coming out in the near-ish future.

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Few recommendations just to make sure that we are covering all the code paths between the setter and the constructor, and a few updates to testing the internal methods.

File: docs/uml-class-diagrams/admin/test/SatelliteTests/SatelliteTests.puml [hunk 4]
```diff
@@ -2,8 +2,21 @@
 
 package edu.oakland.test.admin {
     class SatelliteTests {
-        
+       + void satNameInIsSatNameOut()
+       + void strengthInIsStrengthOut()
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/admin/test/SatelliteTests/SatelliteTests.puml]: This is redundant and can be removed now that you have the two tests below.
```suggestion
```

File: docs/uml-class-diagrams/admin/test/SatelliteTests/SatelliteTests.puml [hunk 5]
```diff
@@ -2,8 +2,21 @@
...
+       + void strengthInIsStrengthOutConstructor()
+       + void strengthInIsStrengthOutSetter()
+       + void getLocation0IsCorrect()
+       + void getLocation1IsCorrect()
+       + void getLocationNIsCorrect()
+       + void initLocationIsNotNull()
+       + void satNameIsNotNull() 
+       + void strengthIsBtwnOneAndTen()
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/admin/test/SatelliteTests/SatelliteTests.puml]: This is redundant and can be removed now that you have the four tests below.
```suggestion
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
One more fix to remove redundant tests, then you're set to merge!

**Review by andrewdimmer — DISMISSED** [review]
Looks good! You're all set to merge as soon as the checks finish!

**Review by andrewdimmer — APPROVED** [review]
Set to merge!

**Review by brendanfraser597 — APPROVED** [review]

**Pull request merged**